addons/hdfs-snapshot-mirroring/src/main/java/org/apache/falcon/replication/snapshots/HdfsSnapshotReplicator.java

```diff
@@ -0,0 +1,305 @@
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.tools.DistCp;
+import org.apache.hadoop.tools.DistCpOptions;
+import org.apache.hadoop.util.Tool;
+import org.apache.hadoop.util.ToolRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * HDFS snapshot generator and snapshot based replicator.
+ */
+public class HdfsSnapshotReplicator extends Configured implements Tool {
+    private static final Logger LOG = LoggerFactory.getLogger(HdfsSnapshotReplicator.class);
+    protected CommandLine cmd;
+
+    public static void main(String[] args) throws Exception {
+        ToolRunner.run(new Configuration(), new HdfsSnapshotReplicator(), args);
+    }
+
+    @Override
+    public int run(String[] args) throws FalconException {
+        cmd = getCommand(args);
+
+        String sourceStorageUrl = cmd.getOptionValue(HdfsSnapshotMirrorProperties.SOURCE_NN.getName());
+        String sourceExecuteEndpoint = cmd.getOptionValue(HdfsSnapshotMirrorProperties.SOURCE_EXEC_URL.getName());
+        String sourcePrincipal = cmd.getOptionValue(
+                HdfsSnapshotMirrorProperties.SOURCE_NN_KERBEROS_PRINCIPAL.getName());
+
+        String targetStorageUrl = cmd.getOptionValue(HdfsSnapshotMirrorProperties.TARGET_NN.getName());
+        String taregtExecuteEndpoint = cmd.getOptionValue(HdfsSnapshotMirrorProperties.TARGET_EXEC_URL.getName());
+        String targetPrincipal = cmd.getOptionValue(
+                HdfsSnapshotMirrorProperties.TARGET_NN_KERBEROS_PRINCIPAL.getName());
+
+        String jobStorageUrl = cmd.getOptionValue(HdfsSnapshotMirrorProperties.JOB_NN.getName());
+        String jobExecuteEndpoint = cmd.getOptionValue(HdfsSnapshotMirrorProperties.JOB_EXEC_URL.getName());
+        String jobPrincipal = cmd.getOptionValue(
+                HdfsSnapshotMirrorProperties.JOB_NN_KERBEROS_PRINCIPAL.getName());
+
+
+        Configuration sourceConf = ClusterHelper.getConfiguration(sourceStorageUrl,
+                sourceExecuteEndpoint, sourcePrincipal);
+        Configuration targetConf = ClusterHelper.getConfiguration(targetStorageUrl,
+                taregtExecuteEndpoint, targetPrincipal);
+        Configuration jobConf = ClusterHelper.getConfiguration(jobStorageUrl, jobExecuteEndpoint, jobPrincipal);
+
+        DistributedFileSystem sourceFs = HadoopClientFactory.get().createDistributedProxiedFileSystem(sourceConf);
+        DistributedFileSystem targetFs = HadoopClientFactory.get().createDistributedProxiedFileSystem(targetConf);
+
+        String currentSnapshotName = "falcon-snapshot-"
+                + cmd.getOptionValue(HdfsSnapshotMirrorProperties.SNAPSHOT_JOB_NAME.getName())
+                + "-" + System.currentTimeMillis();
```

Comment:
"-" is only used at one location in the code. I dont think it makes sense to create a constant for this.